### PR TITLE
First Commit :-) ShooterSubsystem

### PR DESCRIPTION
**Pull request by Evan63w**
ShooterSubsystem

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class ShooterSubsystem extends SubsystemBase {
+	/**
+	 * Creates a new ShooterSubsystem.
+	 */
+	SpeedController m_rightWheel, m_leftWheel, m_angleWheel;
+	Encoder m_rightEncoder, m_leftEncoder, m_angleEncoder;
```
Comment (RafaelPiloto10) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Make private

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+	double rightShooterSpeed = .1;
+	double leftShooterSpeed = .1;
```
Comment (RafaelPiloto10) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Make private

Comment (RafaelPiloto10) [reply src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Should spin in opposite directions

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	public void calculateSpeeds() {
+		// to calculate the different speeds of the right and left motors based on angle
+	}
+
+	public void fire() {
+		m_rightWheel.set(rightShooterSpeed);
+		m_leftWheel.set(leftShooterSpeed);
+	}
```
Comment (RafaelPiloto10) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Don't use a preset "speed". Allow the speeds to be passed in as parameters for future PID Control

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		m_leftWheel.stopMotor();
+	}
+
+	public void goDown() {
+		m_angleWheel.set(-angleWheelSpeed);
+	}
+
+	public void goUp() {
+		m_angleWheel.set(angleWheelSpeed);
+	}
```
Comment (RafaelPiloto10) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Pass in the speeds as parameters in order to have PID Support

**Review by RafaelPiloto10 — CHANGES_REQUESTED** [review]
Make sure to have parameter speeds and add private visibility to some variables.

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java (hunk 1 above)
Comment (Evan63w) [reply src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: implemented

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java (hunk 2 above)
Comment (Evan63w) [reply src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: implemented

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java (hunk 3 above)
Comment (Evan63w) [reply src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: implemented

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 5]
```diff
@@ -8,22 +8,22 @@
 package frc.robot.subsystems;
 
 import edu.wpi.first.wpilibj.Encoder;
-import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.PWMTalonSRX;
```
Comment (RafaelPiloto10) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Should not assume `SpeedController`. Remove this import

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 6]
```diff
@@ -8,22 +8,22 @@
...
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 public class ShooterSubsystem extends SubsystemBase {
 	/**
 	 * Creates a new ShooterSubsystem.
 	 */
-	SpeedController m_rightWheel, m_leftWheel, m_angleWheel;
-	Encoder m_rightEncoder, m_leftEncoder, m_angleEncoder;
+	private PWMTalonSRX m_rightWheel, m_leftWheel, m_angleWheel;
```
Comment (RafaelPiloto10) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Should be a `SpeedController`. Do not assume the motor controller

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 7]
```diff
@@ -8,22 +8,22 @@
...
-	double rightShooterSpeed = .1;
-	double leftShooterSpeed = .1;
+	private double rightShooterSpeed = .1;
+	private double leftShooterSpeed = -.1;
 	private final double angleWheelSpeed = .1;
 
-	public ShooterSubsystem(SpeedController rightWheel, SpeedController leftWheel, SpeedController angleWheel,
-			Encoder rightEncoder, Encoder leftEncoder, Encoder angleEncoder) {
+	public ShooterSubsystem(PWMTalonSRX rightWheel, PWMTalonSRX leftWheel, PWMTalonSRX angleWheel, Encoder rightEncoder,
+			Encoder leftEncoder, Encoder angleEncoder) {
```
Comment (RafaelPiloto10) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Replace `PWMTalonSRX` with `SpeedController`

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 8]
```diff
@@ -8,22 +8,22 @@
...
-	SpeedController m_rightWheel, m_leftWheel, m_angleWheel;
-	Encoder m_rightEncoder, m_leftEncoder, m_angleEncoder;
+	private PWMTalonSRX m_rightWheel, m_leftWheel, m_angleWheel;
+	private Encoder m_rightEncoder, m_leftEncoder, m_angleEncoder;
 
-	double rightShooterSpeed = .1;
-	double leftShooterSpeed = .1;
+	private double rightShooterSpeed = .1;
+	private double leftShooterSpeed = -.1;
 	private final double angleWheelSpeed = .1;
```
Comment (RafaelPiloto10) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Delete these speeds - Not needed since passed as parameters

**Review by RafaelPiloto10 — CHANGES_REQUESTED** [review]
Make the changes suggested in the comments

**Review by RafaelPiloto10 — APPROVED** [review]
Changes were added

**Pull request merged**